### PR TITLE
[Keyboard]KPrepublic bm80v2 Keyboard ISO support 

### PR DESCRIPTION
**Pull request by edwardslau**
## Description
BM80 V2 ISO layout

## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm80v2_iso/config.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#define USB_POLLING_INTERVAL_MS 1
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2_iso/config.h:44]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/kprepublic/bm80v2_iso/config.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
+#define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#define DISABLE_RGB_MATRIX_RAINDROPS
+#define DISABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2_iso/config.h:50]:
```suggestion
```

These are no longer used.

File: keyboards/kprepublic/bm80v2_iso/config.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#define DRIVER_ADDR_1 0b0110000
+#define DRIVER_ADDR_2 0b0110000 // this is here for compliancy reasons.
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 88
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+#define DRIVER_INDICATOR_LED_TOTAL 0
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2_iso/config.h:56]:
```suggestion
```

File: keyboards/kprepublic/bm80v2_iso/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2_iso/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kprepublic/bm80v2_iso/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,41 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2_iso/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kprepublic/bm80v2_iso/rules.mk [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2_iso/rules.mk:21]:
```suggestion
NKRO_ENABLE = yes              # Enable N-Key Rollover
BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by edwardslau** [comment]
Modified the files as requested

**Review by drashna — APPROVED** [review]

File: keyboards/kprepublic/bm80v2_iso/readme.md [hunk 7]
```diff
@@ -0,0 +1 @@
+# bm80v2_iso
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2_iso/readme.md:1]: This needs to be filled out, RE:
https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**